File: jscript/filelist.js
//タイムスタンプ一覧(CSV)を自動生成する.

// ベースファイルパス
var baseDir;
// 対象外とする拡張子のパターン
var excludeExtensionPattern;
// 対象外とするファイル名のパターン
var excludeFilePattern;

var shell;
var fileSystem;


function main() {
	
	shell = WScript.CreateObject("WScript.Shell");
	fileSystem = WScript.CreateObject("Scripting.FileSystemObject");
	
	baseDir = WScript.Arguments(0);

	excludeExtensionPattern = new  RegExp(shell.ExpandEnvironmentStrings("%EXCLUDE_EXTENSION%"));
	excludeFilePattern = new  RegExp(shell.ExpandEnvironmentStrings("%EXCLUDE_FILE%"));

	parse(baseDir);
}

function parse(nowDir) {
	
	var dir = fileSystem.GetFolder(nowDir);
	var files = new Enumerator(dir.files);
	for(; !files.atEnd(); files.moveNext()) {
		var file = files.item();
		if( file.name.search(excludeExtensionPattern) == -1 && 
			file.name.search(excludeFilePattern) == -1 ) {
			printFile( nowDir, file, file.path );
		}
	}
	
	var dirList = dir.subFolders;
	var dirs = new Enumerator(dir.subFolders)
	
	for(; !dirs.atEnd(); dirs.moveNext()) {
		parse(nowDir + "\\" + dirs.item().name);
	}
	
}

function printFile(nowDir, aFile, fullPath) {

	var updateTime = new Date(aFile.DateLastModified);
	var updateTimeStr = updateTime.toLocaleDateString() + " " +  updateTime.toLocaleTimeString();
	WScript.echo(nowDir + "," + aFile.name + "," + aFile.Size + "," +  updateTimeStr);
}

main();

File: ruby/filelist.rb
# 指定フォルダ内のファイルのタイムスタンプ一覧(CSV)を自動生成する.
require 'pathname'
require 'kconv'

#ファイルパスを取得する.
$baseDir = ARGV[0]
$excludeExtensionPattern
$excludeFilePattern

def main
	
	excludeExtension = ENV['EXCLUDE_EXTENSION'].toutf8
	excludeFile = ENV['EXCLUDE_FILE'].toutf8
	
	$excludeExtensionPattern =  /#{excludeExtension}/
	$excludeFilePattern = /#{excludeFile}/
	
	parse($baseDir)
	
end

def parse(nowDir)

	nowDir = nowDir.gsub("/", "\\")
	dirList = Array.new

	Dir.foreach(nowDir) do |aPath|
		if aPath != "." && aPath != ".." then
			fullPath = (Pathname.new(nowDir) + aPath).to_s
			ftype = File.ftype(fullPath)
			if ftype == "file" then
				if not aPath.toutf8 =~ $excludeExtensionPattern  and
				   not aPath.toutf8 =~ $excludeFilePattern then
				   printFile( nowDir, aPath, fullPath )
				end
			else ftype == "directory"
				dirList.push(aPath)
			end
		end
	end
	
	dirList.each() do |aDir|
		fullPath = (Pathname.new(nowDir) + aDir).to_s
		parse(fullPath)
	end
end


def printFile(nowDir, aFilename, fullPath)
	#出力.
	updateTime = File.mtime(fullPath)
	updateTimeStr = updateTime.strftime("%Y/%m/%d %H:%M:%S")
	puts nowDir + "," + aFilename + "," + File.size(fullPath).to_s + ", " +  updateTimeStr
end

main